checkov/common/runners/runner_registry.py

```diff
@@ -99,3 +114,67 @@ def enrich_report_with_guidelines(scan_report, guidelines):
+
+    @staticmethod
+    def get_enriched_resources(report, repo_root):
+        tf_definitions = {}
+        parsing_errors = {}
+        Parser().parse_directory(
+            directory=repo_root,  # assume plan file is in the repo-root
+            out_definitions=tf_definitions,
+            out_parsing_errors=parsing_errors,
+        )
+
+        enriched_resources = defaultdict(dict)
+        for full_file_path, definition in tf_definitions.items():
+            definitions_context = parser_registry.enrich_definitions_context((full_file_path, definition))
+            abs_scanned_file, _ = tf_runner._strip_module_referrer(full_file_path)
+            scanned_file = os.path.relpath(abs_scanned_file, repo_root)
+            for block_type, block_value in definition.items():
+                if block_type in CHECK_BLOCK_TYPES:
+                    for entity in block_value:
+                        context_parser = parser_registry.context_parsers[block_type]
+                        definition_path = context_parser.get_entity_context_path(entity)
+                        entity_id = ".".join(definition_path)
+                        entity_context_path = [block_type] + definition_path
+                        entity_context = dict_utils.getInnerDict(
+                            definitions_context[full_file_path], entity_context_path
+                        )
+                        entity_lines_range = [
+                            entity_context.get("start_line"),
+                            entity_context.get("end_line"),
+                        ]
+                        entity_code_lines = entity_context.get("code_lines")
+                        skipped_checks = entity_context.get("skipped_checks")
+                        enriched_resources[entity_id] = {
+                            "entity_code_lines": entity_code_lines,
+                            "entity_lines_range": entity_lines_range,
+                            "scanned_file": scanned_file,
+                            "skipped_checks": skipped_checks
+                        }
+        return enriched_resources
+
+    @staticmethod
+    def enrich_plan_report(report, enriched_resources):
+        # This enriches reports with the appropriate filepath, line numbers, and codeblock
+        for record in report.failed_checks:
+            enriched_resource = enriched_resources.get(record.resource)
+            if enriched_resource:
+                record.file_path = enriched_resource["scanned_file"]
+                record.file_line_range = enriched_resource["entity_lines_range"]
+                record.code_block = enriched_resource["entity_code_lines"]
+        return report
+
+    @staticmethod
+    def handle_skipped_checks(report, enriched_resources):
+        records_to_skip = []
+        for record in report.failed_checks:
+            resource_skips = enriched_resources[record.resource]["skipped_checks"]
```

Comment:
This can result in an exception, even using `defaultdict`. If the key `record.resource` doesn't exist, then `defaultdict` would return a dict, but then accessing `skipped_checks` key will raise a `KeyError`.